main.cpp

```diff
@@ -125,151 +182,149 @@ struct CarWash
  */
 
-
-/*
-Thing 1) Coffee Shop
-5 properties:
-    1) number of chairs (int)
-    2) number of staff (int)
-    3) number of toilets (int)
-    4) annual profit (double)
-    5) price of coffee (float)
-3 things it can do:
-    1) host local artists work on walls
-    2) charge customer
-    3) train staff
-*/
-
 struct CoffeeShop
 {
-
-    //number of chairs (int)
     int numChairs = 54;
-    //number of staff (int)
     int numStaff = 7;
-    //number of toilets (int)
     int numToilets = 2;
-    //annual profit (double)
     double annualProfit = 45362.78;
-    //price of coffee (float)
     float priceCoffee = 3.40f;
 
     struct Employee
     {
         std::string firstName = "Magnus";
         std::string lastName = "Magnusson";
         int employeeID = 12345;
-        bool fullyTrained = true;
+        bool fullyTrained = false;
         float salary = 30000.00f;
 
-        void makeDeliveries();
-        void takeHoliday();
-        bool deservesPromotion(); // returns whether the employee is derving of a promotion
+        void makeDeliveries(float distTravelled, int numCustomers);
+        void takeHoliday(int startDate, int endDate);
+        bool deservesPromotion();
     };
 
-    //host local artists work on walls
-    float hostLocalArtistOnWalls(float percenatgeCommisionAgreed, float totalArtSales, CoffeeShop coffeeshop); //returns the total value of commission earn from the sale of all the art at this instance of CoffeeShop 
-    //charge customer
-    float chargeCustomer(float priceCoffee, int numberOfCoffees, int customerNumber); //returns the total value of the sale of all good for this customer
-    //train staff
+    float hostLocalArtistOnWalls(float percenatgeCommissionAgreed, float totalArtSales, std::string name);
+    float chargeCustomer(int numberOfCoffees, std::string customerName);
     void trainStaff(Employee employee, std::string trainingRequired);
 };
+void CoffeeShop::Employee::makeDeliveries(float distTravelled, int numCustomers)
+{
+    float aveDistancePerCustomer = distTravelled / numCustomers;
+    std::cout << aveDistancePerCustomer;
+}
 
-/*
-Thing 2) School
-5 properties:
-    1) number of classrooms (int)
-    2) position in leauge table (int)
-    3) number of teachers (int)
-    4) number of pupils (int)
-    5) student average grade (float)
-3 things it can do:
-    1) train teachers
-    2) produce students christmas play
-    3) close for holidays
-*/
+void CoffeeShop::Employee::takeHoliday(int startDate, int endDate)
+{
+    int numDaysHolidayTaken = endDate - startDate;
+    std::cout << "Days holiday taken by " << firstName << " " << lastName << numDaysHolidayTaken << std::endl;
+}
+bool CoffeeShop::Employee::deservesPromotion()
+{
+    if (fullyTrained)
+    {
+        salary += 1000;
+        fullyTrained = true;
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise